### PR TITLE
Kernel+LibC: Add the futimens syscall, properly implement the LibC futimens function

### PR DESCRIPTION
**Pull request by supercomputer7**
This fixes an issue with the LibC futimens function, which was very easy to trigger - simply run "touch new-file-which-not-existing" and you could see that even though the file is created, you will get an error in `stderr`.

### CODE REVIEW

File: Kernel/Syscalls/utimensat.cpp [hunk 1]
```diff
@@ -12,6 +12,40 @@
...
+    }
+
+    auto description = TRY(open_file_description(params.fd));
+    if (!description->inode())
+        return EBADF;
+    if (!description->custody())
+        return EBADF;
+
+    auto& atime = times[0];
+    auto& mtime = times[1];
```
Comment (konradekk) [inline Kernel/Syscalls/utimensat.cpp:44]: is this table necessary?

**Review by konradekk — APPROVED** [review]

File: Kernel/Syscalls/utimensat.cpp (hunk 1 above)
Comment (supercomputer7) [reply Kernel/Syscalls/utimensat.cpp:44]: Well the old code used this array, so I guess there's nothing wrong with this.

**Comment by awesomekling** [comment]
The syscall name `sys$futimensat` is inaccurate as the `at` suffix implies the presence of a "directory fd" parameter, but no such parameter is being passed here.

Please name is `sys$futimens` to match the LibC API (and the POSIX specification). :)

**Comment by supercomputer7** [comment]
> The syscall name `sys$futimensat` is inaccurate as the `at` suffix implies the presence of a "directory fd" parameter, but no such parameter is being passed here.
> 
> Please name is `sys$futimens` to match the LibC API (and the POSIX specification). :)

Done.

File: Userland/Libraries/LibC/fcntl.h [hunk 2]
```diff
@@ -32,6 +32,7 @@ int inode_watcher_remove_watch(int fd, int wd);
 int posix_fadvise(int fd, off_t offset, off_t len, int advice);
 int posix_fallocate(int fd, off_t offset, off_t len);
 
+int __utimens(int fd, char const* path, struct timespec const times[2], int flag);
```
Comment (ADKaster) [inline Userland/Libraries/LibC/fcntl.h:35]: We don't want to expose this helper to users of LibC do we? It should be file static.

Comment (supercomputer7) [reply Userland/Libraries/LibC/fcntl.h]: Resolved in a discussion in the discord server (https://discord.com/channels/830522505605283862/830525093905170492/1094800326172889129).

**Review by ADKaster — APPROVED** [review]

**Pull request merged**